gtsam/hybrid/HybridGaussianFactor.cpp

```diff
@@ -73,22 +66,65 @@ static HybridGaussianFactor::Factors augment(
     return std::dynamic_pointer_cast<GaussianFactor>(
         std::make_shared<JacobianFactor>(gfg));
   };
-  return HybridGaussianFactor::Factors(factors, update);
+  return Factors(factors, update);
 }
 
 /* *******************************************************************************/
-HybridGaussianFactor::HybridGaussianFactor(const KeyVector &continuousKeys,
-                                           const DiscreteKeys &discreteKeys,
-                                           const FactorValuePairs &factors)
-    : Base(continuousKeys, discreteKeys), factors_(augment(factors)) {}
+HybridGaussianFactor::ConstructorHelper::ConstructorHelper(
+    const DiscreteKey &discreteKey,
+    const std::vector<GaussianFactor::shared_ptr> &factors)
+    : discreteKeys({discreteKey}) {
+  // Extract continuous keys from the first non-null factor
```

Comment:
I'll do a full review in a bit, but I am wondering about the extra compute happening here when we could have just given the keys directly.

Currently I am struggling with the hybrid state estimator being far too slow and having too many hypotheses. Pruning is working but the blow up is still immense, hence my concerns here.

Reply:
Ironically, it's a bit faster.

Reply:
Esp, GaussianConditional, which did *two* tree traversals, now only does one.